File: Competitions/UC/calico/contest/unlockmanifolds/templates/unlockmanifolds.cpp
#include <iostream>
#include <vector>

using namespace std;

/**
 * Return the minimum number of actions
 *
 * N: a non-negative integer representing the number of rows
 * M: another non-negative integer representing the number of columns
 * G: N x M array representing a grid
 */
int solve(int N, int M, vector<vector<int>> &G) {
    // YOUR CODE HERE
    return -1;
}

int main()
{
    int T;
    cin >> T;
    for (int i = 0; i < T; i++) {
        int N, M;
        cin >> N >> M;
        vector<vector<int>> G(N, vector<int>(M));
        for (int j = 0; j < N; j++) {
            for (int k = 0; k < M; k++) {
                cin >> G[j][k];
            }
        }
        cout << solve(N, M, G) << '\n';
    }
}

File: Competitions/UC/calico/contest/unlockmanifolds/templates/unlockmanifolds.py
def solve(N: int, M: int, G: list) -> int:
    """
    Return the minimum number of actions
    
    N: a non-negative integer representing the number of rows
    M: another non-negative integer representing the number of columns
    G: N x M array representing a grid
    """
    # YOUR CODE HERE
    return -1


def main():
    T = int(input())
    for _ in range(T):
        N, M = map(int, input().split())
        G = []
        for _ in range(N):
            row = list(map(int, input().split()))
            G.append(row)
        print(solve(N, M, G))


if __name__ == '__main__':
    main()
